src/sentry/attachments/base.py

```diff
@@ -25,60 +48,112 @@ def from_upload(cls, file, **kwargs):
 
     @property
     def data(self):
-        if self._data is None and self._load is not None:
-            self._data = self._load()
+        if self._data is None and self._cache is not None:
+            assert self.id is not None
+
+            if self.chunks is None:
+                self._data = self._cache.get_unchunked_data(key=self.key, id=self.id)
```

Comment:
Can we combine both cases, and just have a single load method? This would allow to mget.

```python
def load(self, key, id, chunks=None):
  if chunks is None:
    # single fetch without chunks
  else:
    # mget on chunk ids, return joined result
```

If this works, then same for `set`, please. Having both `get_chunk` and `set_chunk` in addition to that on the cache is still a good idea (and required for the consumer, iirc).

Reply:
I added a `get_data` that takes an attachment and refactored `delete` to use `get`.

I don't know how to make `set` work that way